test/testdetail/TestMacrosImpl.hh

```diff
@@ -724,6 +727,81 @@ template<class ContainerE, class ContainerA, class T>
         BinaryOp{static_cast<ValueT>(rel), static_cast<ValueT>(abs)});
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Compare two vectors of reference values using a tolerance.
+ */
+template<class ContainerE, class ContainerA, class Tol>
+std::enable_if_t<IsContainer<ContainerE>::value && IsContainer<ContainerA>::value,
+                 ::testing::AssertionResult>
+IsRefEq(char const* expr1,
+        char const* expr2,
+        char const* tol_expr,
+        ContainerE const& val1,
+        ContainerA const& val2,
+        Tol const& tol)
+{
+    ::celeritas::test::AssertionHelper result{expr1, expr2};
+    using std::begin;
+    using std::end;
+
+    if (result.equal_size(std::distance(begin(val1), end(val1)),
+                          std::distance(begin(val2), end(val2))))
+    {
+        // Check each element
+        auto iter1 = begin(val1);
+        auto iter2 = begin(val2);
+        size_type i = 0;
+        size_type failures = 0;
+        constexpr size_type max_printable_failures = 10;
+        for (auto end1 = end(val1); iter1 != end1; ++i, ++iter1, ++iter2)
+        {
+            ::testing::AssertionResult item_result
+                = ::testing::AssertionSuccess();
+            if constexpr (!std::is_same_v<Tol, std::nullptr_t>)
+            {
+                // Compare with tolerance
+                item_result
+                    = IsRefEq(expr1, expr2, tol_expr, *iter1, *iter2, tol);
+            }
+            else
+            {
+                item_result = IsRefEq(expr1, expr2, *iter1, *iter2);
+            }
+            if (!item_result)
+            {
+                if (failures++ < max_printable_failures)
+                {
+                    result.fail() << item_result << "\n(Failed in element "
+                                  << i << " of " << expr2 << ")";
```

Comment:
Should we also mention `expr1` here?

Reply:
It should be the reference and is long, so I think better to omit to avoid confusion. It often shows the diff anyway. This is mostly for facilitating copy paste to update reference values.